src/server/game/Entities/Player/Player.cpp

```diff
@@ -14122,11 +14122,22 @@ void Player::ActivateSpec(uint8 spec)
     }
 
     // xinef: apply glyphs from second spec
-    if(GetActiveSpec() != oldSpec)
+    if (GetActiveSpec() != oldSpec)
+    {
         for (uint8 slot = 0; slot < MAX_GLYPH_SLOT_INDEX; ++slot)
-            if (uint32 glyphId = m_Glyphs[GetActiveSpec()][slot])
+        {
+            uint32 glyphId = m_Glyphs[GetActiveSpec()][slot];
```

Comment:
if you put this `m_Glyphs[GetActiveSpec()]` outside in another variable you avoid to call at any iteration the `GetActiveSpec()` function, up to you I already approved the PR

Reply:
```
uint32 m_Glyphs[MAX_TALENT_SPECS][MAX_GLYPH_SLOT_INDEX];
```
So no - it's not possible in that way :)